### ISSUE TITLE
False positive for runtime/explicit on deleted constructors

### ISSUE THREAD

**Issue by ptomato** [cpplint/cpplint#386]
Deleted constructors shouldn't need to be marked explicit, because they can't be called implicitly.

Test case: **explicit.cpp**
```c++
// Copyright 2025 The Public Domain
class A {};

class B {
  B(A) = delete;
};
```

Actual output:
```
$ cpplint --version
Cpplint fork (https://github.com/cpplint/cpplint)
cpplint 2.0.2
Python 3.13.5 (main, Jun 12 2025, 00:00:00) [GCC 15.1.1 20250521 (Red Hat 15.1.1-2)]
$ cpplint explicit.cpp
explicit.cpp:5:  Single-parameter constructors should be marked explicit.  [runtime/explicit] [4]
Done processing explicit.cpp
Total errors found: 1
```
